.circleci/config.yml

```diff
@@ -0,0 +1,155 @@
+---
+version: 2.1
+
+references:
+  images:
+    go: &GOLANG_IMAGE docker.mirror.hashicorp.services/circleci/golang:1.15.3
+    middleman: &MIDDLEMAN_IMAGE docker.mirror.hashicorp.services/hashicorp/middleman-hashicorp:0.3.44
+    ember: &EMBER_IMAGE docker.mirror.hashicorp.services/circleci/node:12-browsers
+
+  paths:
+    test-results: &TEST_RESULTS_DIR /tmp/test-results
+
+  cache:
+    yarn: &YARN_CACHE_KEY waypoint-ui-v1-{{ checksum "ui/yarn.lock" }}-{{ checksum "uilib.md5" }}
+    rubygem: &RUBYGEM_CACHE_KEY static-site-gems-v1-{{ checksum "Gemfile.lock" }}
+
+  environment: &ENVIRONMENT
+    TEST_RESULTS_DIR: *TEST_RESULTS_DIR
+    GOTESTSUM_RELEASE: 0.4.2
+    EMAIL: noreply@hashicorp.com
+    GIT_AUTHOR_NAME: circleci-waypoint
+    GIT_COMMITTER_NAME: circleci-waypoint
+    BASH_ENV: /home/circleci/project/.circleci/bash_env.sh
+    DOCKER_BUILDKIT: 1
+    REBUILD_UI: &REBUILD_UI 1
+
+jobs:
+  # Runs Go linters
+  lint:
+    docker:
+      - image: *GOLANG_IMAGE
+    environment:
+      GOTAGS: ""
+    steps:
+      - add_ssh_keys:
+          fingerprints:
+            - "c6:96:98:82:dc:04:6c:39:dd:ac:83:05:e3:15:1c:98"
+      - checkout
+      - run:
+          name: Install golangci-lint
+          command: |
+            download=https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh
+            wget -O- -q $download | sh -x -s -- -d -b /go/bin/ v1.27.0
+      - run: go mod download
+      - run:
+          name: lint
+          command: &lintcmd |
+            golangci-lint run --build-tags="$GOTAGS" -v --concurrency 2 \
+              --disable-all \
+              --timeout 10m \
+              --enable gofmt \
+              --enable gosimple \
+              --enable govet
+      - notify_main_failure
+
+  # checks vendor directory is correct
+  check-vendor:
+    docker:
+      - image: *GOLANG_IMAGE
+    environment:
+      <<: *ENVIRONMENT
+    steps:
+      - add_ssh_keys:
+          fingerprints:
+            - "c6:96:98:82:dc:04:6c:39:dd:ac:83:05:e3:15:1c:98"
+      - checkout
+      - run:
+          command: go mod tidy
+      - run: |
+          if ! git diff --exit-code; then
+            echo "Git directory has vendor changes"
+            exit 1
+          fi
+      - notify_main_failure
+
+  go-test:
+    docker:
+      - image: *GOLANG_IMAGE
+      - image: docker.mirror.hashicorp.services/localstack/localstack:latest
+      - image: docker.mirror.hashicorp.services/letsencrypt/pebble
+        command: ["pebble"]
+        environment:
+          PEBBLE_VA_NOSLEEP: 1
+          PEBBLE_VA_ALWAYS_VALID: 1
+      - image: docker.mirror.hashicorp.services/circleci/postgres:11-alpine
+        environment:
+          POSTGRES_USER: postgres
+          POSTGRES_DB: waypoint_test
+      - image: docker.mirror.hashicorp.services/vault
+        command: ["server", "-dev", "-dev-root-token-id=hznroot"]
```

Comment:
I don't htink we need to worry about postgres or horizon on the sdk side?

Reply:
Yep agreed....I missed some stuff from the copy over from Waypoint! :smile: